src/mixer.rs

```diff
@@ -0,0 +1,428 @@
+//! Mixer API - Simple Mixer API for mixer control
+//!
+use std::ffi::CString;
+use std::{ptr, mem};
+
+use alsa;
+use super::card;
+use super::error::*;
+
+const SELEM_ID_SIZE: usize = 64;
+
+/// wraps [snd_mixer_t](http://www.alsa-project.org/alsa-doc/alsa-lib/group___mixer.html)
+pub struct Mixer(*mut alsa::snd_mixer_t);
+
+impl Mixer {
+    /// Creates a new iterator for a specific card using the cards index and
+    /// using hw name, i.e. `hw:0`
+    pub fn new(c: &card::Card) -> Result<Mixer> {
+        let card = &CString::new(format!("hw:{}", c.get_index())).unwrap();
+        let mut mixer_handle = ptr::null_mut();
+        try!(acheck!(snd_mixer_open(&mut mixer_handle, 0)));
+        try!(acheck!(snd_mixer_attach(mixer_handle, card.as_ptr())));
```

Comment:
Need to put handle into Mixer already here - as long as you've done snd_mixer_open, you must do snd_mixer_close or you'll leak the handle